File: app/src/main/java/com/funckyhacker/fileexplorer/view/main/MainView.java
package com.funckyhacker.fileexplorer.view.main;

import android.content.Intent;
import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter;

public interface MainView {

  void setAdapter(MainLinearAdapter adapter);

  void startActivity(Intent intent);

  void showSnackBar(String message);
}

File: app/src/main/java/com/funckyhacker/fileexplorer/view/main/MainView.kt
package com.funckyhacker.fileexplorer.view.main

import android.content.Intent
import android.support.annotation.StringRes
import com.funckyhacker.fileexplorer.view.adapter.MainLinearAdapter

interface MainView {

    fun setAdapter(adapter: MainLinearAdapter)

    fun startActivity(intent: Intent)

    fun showSnackBar(message: String)

    fun showErrorDialog(@StringRes messageId: Int)
}
